service/src/main/java/bio/terra/cbas/dependencies/bard/BardService.java

```diff
@@ -0,0 +1,82 @@
+package bio.terra.cbas.dependencies.bard;
+
+import bio.terra.bard.api.DefaultApi;
+import bio.terra.bard.client.ApiClient;
+import bio.terra.bard.model.EventsEvent200Response;
+import bio.terra.bard.model.EventsEventLogRequest;
+import bio.terra.cbas.dependencies.common.HealthCheck;
+import bio.terra.cbas.model.RunSetRequest;
+import bio.terra.cbas.models.GithubMethodDetails;
+import bio.terra.cbas.models.MethodVersion;
+import bio.terra.common.iam.BearerToken;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+@Component
+public class BardService implements HealthCheck {
+
+  Logger log = LoggerFactory.getLogger(BardService.class);
+
+  private final BardClient bardClient;
+  private final String appId = "cbas";
+
+  public BardService(BardClient bardClient) {
+    this.bardClient = bardClient;
+  }
+
+  private DefaultApi getDefaultApi(BearerToken userToken) {
+    ApiClient client = bardClient.bardAuthClient(userToken);
+    return bardClient.defaultApi(client);
+  }
+
+  public void logRunSetEvent(
```

Comment:
I worry about this by synchronous. If Bard, for some reason, is slow in responding, it'll slow down CBAS. [In DRSHub](https://github.com/DataBiosphere/terra-drs-hub/blob/068e2389d3bbd9b1bc888bb574c5ca096d82b5f6/service/src/main/java/bio/terra/drshub/services/TrackingService.java#L34), logging to Bard is async, so that it happens in a different threadpool. We should probably do the same thing here.

Reply:
Saloni has a PR to add an async work queue so we could wait for that to go in first and use that

Reply:
I think that would be a a good idea

Reply:
Saloni's PR is now merged. Do you feel like you're willing to have a go at making this async with the same mechanism? We might want a separate queue for this, but it should be easy enough to make that work